typescript/cli/src/deploy/warp.ts

```diff
@@ -474,17 +426,34 @@ async function getOrDeployIsmFactories(
   registry: IRegistry,
   chain: string,
   ismFactoryDeployer: HyperlaneProxyFactoryDeployer,
-) {
+): Promise<HyperlaneAddresses<ProxyFactoryFactories>> {
   logBlue(`Loading registry factory addresses for ${chain}...`);
   let chainAddresses = await registry.getChainAddresses(chain);
 
   if (!chainAddresses) {
     logGray(`Registry factory addresses not found for ${chain}. Deploying...`);
     chainAddresses = serializeContracts(
       await ismFactoryDeployer.deployContracts(chain),
-    ) as Record<string, string>;
+    );
   }
-  return chainAddresses;
+
+  return {
+    staticAggregationHookFactory: chainAddresses!.staticAggregationHookFactory,
+    staticAggregationIsmFactory: chainAddresses!.staticAggregationIsmFactory,
+    staticMerkleRootMultisigIsmFactory:
+      chainAddresses!.staticMerkleRootMultisigIsmFactory,
+    staticMessageIdMultisigIsmFactory:
+      chainAddresses!.staticMessageIdMultisigIsmFactory,
+    domainRoutingIsmFactory: chainAddresses!.domainRoutingIsmFactory,
+
+    // These can be null because they may not exist on all chains in the Registry
```

Comment:
Do we have a ticket tracking work to deploy to this to all chains?

Reply:
Yup fixed by @aroralanuk